<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>uactestClass</class>
 <widget class="QMainWindow" name="uactestClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>935</width>
    <height>784</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>UAC Test</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="calcButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>221</width>
      <height>131</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;FOR SCIENCE!&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Calculate!</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="cooldownBox">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>20</y>
      <width>271</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Cooldown Modifiers</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QCheckBox" name="fastfireCheck">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enables Fast Fire.&lt;/p&gt;&lt;p&gt;This results in 5% CDR.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Fast Fire (+5%)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="moduleCheck">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enables UAC Cooldown Module.&lt;/p&gt;&lt;p&gt;This assumes a Rank 5 module, which results in 12% CDR.&lt;/p&gt;&lt;p&gt;(Note: disabled for UAC2)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>CD Module (+12%)</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="loadoutBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>311</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Loadout</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0">
     <item>
      <layout class="QHBoxLayout" name="loadoutLayout" stretch="1,2">
       <item>
        <layout class="QVBoxLayout" name="uacSelectLayout" stretch="0,0">
         <item>
          <widget class="QLabel" name="uacSelectLabel">
           <property name="text">
            <string>UAC:</string>
           </property>
           <property name="buddy">
            <cstring>uacSelect</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="uacSelect">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;UAC to test.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <item>
            <property name="text">
             <string/>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>20</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="cycleCountLayout" stretch="0,0">
         <item>
          <widget class="QLabel" name="cycleCountLabel">
           <property name="text">
            <string>Cycles</string>
           </property>
           <property name="buddy">
            <cstring>cycleCount</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="cycleCount">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Number of double-tap cycles to be tested over.&lt;/p&gt;&lt;p&gt;Min: 10000 (ensures accuracy)&lt;/p&gt;&lt;p&gt;Max: 100000000 (10&lt;span style=&quot; vertical-align:super;&quot;&gt;8&lt;/span&gt;; you may not want to go over 10&lt;span style=&quot; vertical-align:super;&quot;&gt;7&lt;/span&gt; anyways)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="minimum">
            <number>10000</number>
           </property>
           <property name="maximum">
            <number>100000000</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>180</y>
      <width>361</width>
      <height>131</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <property name="labelAlignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="timeDisplayLabel">
       <property name="text">
        <string>Total Time:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="timeDisplay">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Total time (in seconds) to complete all cycles, assuming perfect timing (no delay between cooldown and firing).&lt;/p&gt;&lt;p&gt;&lt;br/&gt;Devnote: due to the way things are coded, it will actually be greater by (Cooldown) seconds, which matters... not much in the long run (10&lt;span style=&quot; vertical-align:super;&quot;&gt;4&lt;/span&gt;+ cycles).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="text">
        <string>time</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="damageDisplayLabel">
       <property name="text">
        <string>Total Damage:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="damageDisplay">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Total damage dealt over all cycles.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="text">
        <string>damage</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="dpsDisplayLabel">
       <property name="text">
        <string>Final DPS:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="dpsDisplay">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Average DPS over all cycles.&lt;/p&gt;&lt;p&gt;More simply: Total Damage / Total Time&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="text">
        <string>dps</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <zorder>calcButton</zorder>
   <zorder>cooldownBox</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>loadoutBox</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>935</width>
     <height>38</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="uactest.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>uacSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>uactestClass</receiver>
   <slot>checkModuleValid(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cycleCount</sender>
   <signal>valueChanged(int)</signal>
   <receiver>uactestClass</receiver>
   <slot>checkCalcReady()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>556</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uacSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>uactestClass</receiver>
   <slot>checkCalcReady()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>559</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>calcButton</sender>
   <signal>pressed()</signal>
   <receiver>uactestClass</receiver>
   <slot>onCalcButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>checkModuleValid(int)</slot>
  <slot>checkCalcReady()</slot>
  <slot>onCalcButtonClicked()</slot>
 </slots>
</ui>
